src/extension/providers/liquid/index.ts

```diff
@@ -1,26 +1,17 @@
-import { PromiseQueue } from "~/extension/providers/promiseQueue";
-import { postMessage } from "../postMessage";
+import CommonProvider from "~/extension/providers/commonProvider";
 
 declare global {
   interface Window {
     liquid: LiquidProvider;
   }
 }
 
-export default class LiquidProvider {
-  enabled: boolean;
-  private _queue: PromiseQueue;
-
-  constructor() {
-    this.enabled = false;
-    this._queue = new PromiseQueue();
-  }
-
+export default class LiquidProvider extends CommonProvider {
   async enable(): Promise<void> {
     if (this.enabled) {
       return;
     }
-    const result = await this.execute("enable");
+    const result = await this.execute("liquid", "enable");
```

Comment:
can the `enable` method also be moved to the base class?

Reply:
resolved! i didn't refactored it because of the way i was using introducing scope in refactored code! replacing the logic, now allow us to refactor this as well. thanks